libs/payments/ui/src/lib/server/components/PurchaseDetails/index.tsx

```diff
@@ -57,17 +58,34 @@ type PurchaseDetailsProps = {
     productName: string;
     webIcon: string;
   };
+  discountEnd?: number | null;
+  discountType?: string;
+  invoiceEnd: number;
 };
 
 export async function PurchaseDetails(props: PurchaseDetailsProps) {
-  const { purchaseDetails, invoice, interval, l10n } = props;
+  const {
+    purchaseDetails,
+    discountEnd,
+    discountType,
+    invoice,
+    invoiceEnd,
+    interval,
+    l10n,
+  } = props;
   const { currency, listAmount, discountAmount, totalAmount, taxAmounts } =
     invoice;
   const { details, subtitle, productName, webIcon } = purchaseDetails;
   const exclusiveTaxRates = taxAmounts.filter(
     (taxAmount) => !taxAmount.inclusive
   );
 
+  let couponEndDate;
+  if (discountEnd) {
+    couponEndDate = l10n.getLocalizedDateString(discountEnd, true);
+  }
+  const planRenewalDate = invoiceEnd + 86400;
```

Comment:
Calculating the `planRenewalDate` is something that should happen on the backend, in my opinion.

This seems like something Stripe related, rather than frontend specific logic.